automation/run-tests.sh

```diff
@@ -283,8 +215,40 @@ function upload_coverage {
     fi
 }
 
+function check_iface_exist {
+    exec_cmd "ip a s | grep -q $1"
+}
+
+function prepare_network_environment {
+    set +e
+    check_iface_exist 'eth1'
+    if [ $? -eq 1 ]; then
+        VETH_ETH1_CREATED="yes"
+        exec_cmd 'ip link add eth1 type veth peer name eth1peer && ip link set eth1peer up'
+	exec_cmd 'nmcli device set eth1 managed yes'
+    fi
+
+    check_iface_exist 'eth2'
+    if [ $? -eq 1 ]; then
+        VETH_ETH2_CREATED="yes"
+        exec_cmd 'ip link add eth2 type veth peer name eth2peer && ip link set eth2peer up'
+	exec_cmd 'nmcli device set eth2 managed yes'
+    fi
+    set -e
+}
+
+function teardown_network_environment {
+    if [ $VETH_ETH1_CREATED == "yes" ]; then
```

Comment:
If you add the devices to a list (see above), then you can also use a loop here.

Reply:
Done, thanks!